src/go/rpk/pkg/cli/cmd/topic/consume.go

```diff
@@ -427,27 +427,30 @@ func unixMilli(nano int64) int64 { return nano / 1e6 }
 // The expressions below all end in (?::|$), meaning, a non capturing group for
 // the first time followed by a colon or the second time at the end.
 func parseConsumeTimestamp(
-	half string,
-) (length int, at time.Time, end bool, err error) {
+	half string, t0 ...time.Time,
+) (length int, at time.Time, end bool, fromT0 bool, err error) {
```

Comment:
`fromT0` might be clearer as `fromTimestamp` -- i.e., if the first half is a timestamp, then if the second half is a duration, we parse relative to the timestamp.

wdyt? It took me a bit to remember why we were parsing from t0, whereas I think it'd be clearer if it says it's from a timestamp.

Reply:
That makes sense.